code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -82,56 +96,124 @@ export const storybookTest = (options?: UserOptions): Plugin => {
         });
       }
 
-      const presets = await loadAllPresets({
-        configDir,
-        corePresets: [],
-        overridePresets: [],
-        packageJson: {},
-      });
-
-      const workingDir = process.cwd();
-      const directories = {
-        configDir,
-        workingDir,
-      };
-      storiesGlobs = await presets.apply('stories');
-      const indexers = await presets.apply('experimental_indexers', []);
-      const docsOptions = await presets.apply<DocsOptions>('docs', {});
-      const normalizedStories = normalizeStories(await storiesGlobs, directories);
-
-      const generator = new StoryIndexGenerator(normalizedStories, {
+      // performance optimization: load all in parallel
+      const [
+        //
+        framework,
+        storiesGlobsData,
+        indexers,
+        docsOptions,
+        storybookEnv,
+        viteConfigFromStorybook,
+        previewLevelTagsData,
+      ] = await Promise.all([
+        //
+        presets.apply('framework', undefined),
+        presets.apply('stories'),
+        presets.apply('experimental_indexers', []),
+        presets.apply<DocsOptions>('docs', {}),
+        presets.apply('env', {}),
+        presets.apply('viteFinal', {}),
+
+        extractTagsFromPreview(configDir),
+      ]);
+
+      const generator = new StoryIndexGenerator(normalizeStories(storiesGlobsData, directories), {
         ...directories,
         indexers: indexers,
         docs: docsOptions,
-        workingDir,
       });
 
       await generator.initialize();
```

Comment:
depending on how much time this takes, you could move it up into the Promise.all list. But not worth saving 10 ms.